### ISSUE TITLE
Null reference exception when executing migration for adding column of type DateTimeOffset? or DateTimeOffset

### ISSUE THREAD

**Issue by lcrumbling** [bubibubi/EntityFrameworkCore.Jet#100]
I added a column to my data structures:

` public DateTimeOffset? colname { get; set }`

And created a migration:

```
public partial class init2 : Migration
{
    protected override void Up(MigrationBuilder migrationBuilder)
    {
        migrationBuilder.AddColumn<DateTimeOffset>(
            name: "colname",
            table: "tablename",
            nullable: true);
    }

    protected override void Down(MigrationBuilder migrationBuilder)
    {
        migrationBuilder.DropColumn(
            name: "colname",
            table: "tablename");
    }
}
```

When I execute the migration step, I get a Null Reference Exception. 
-2147467261
 Object reference not set to an instance of an object.

StackTrace is:

>    at Microsoft.EntityFrameworkCore.Migrations.MigrationsSqlGenerator.GetColumnType(String schema, String table, String name, ColumnOperation operation, IModel model)
>    at Microsoft.EntityFrameworkCore.Migrations.JetMigrationsSqlGenerator.GetColumnType(String schema, String table, String name, ColumnOperation operation, IModel model)
>    at Microsoft.EntityFrameworkCore.Migrations.MigrationsSqlGenerator.ColumnDefinition(String schema, String table, String name, ColumnOperation operation, IModel model, MigrationCommandListBuilder builder)
>    at Microsoft.EntityFrameworkCore.Migrations.MigrationsSqlGenerator.ColumnDefinition(AddColumnOperation operation, IModel model, MigrationCommandListBuilder builder)
>    at Microsoft.EntityFrameworkCore.Migrations.MigrationsSqlGenerator.Generate(AddColumnOperation operation, IModel model, MigrationCommandListBuilder builder, Boolean terminate)
>    at Microsoft.EntityFrameworkCore.Migrations.JetMigrationsSqlGenerator.Generate(AddColumnOperation operation, IModel model, MigrationCommandListBuilder builder, Boolean terminate)
>    at Microsoft.EntityFrameworkCore.Migrations.MigrationsSqlGenerator.<>c.<.cctor>b__71_0(MigrationsSqlGenerator g, MigrationOperation o, IModel m, MigrationCommandListBuilder b)
>    at Microsoft.EntityFrameworkCore.Migrations.MigrationsSqlGenerator.Generate(MigrationOperation operation, IModel model, MigrationCommandListBuilder builder)
>    at Microsoft.EntityFrameworkCore.Migrations.JetMigrationsSqlGenerator.Generate(MigrationOperation operation, IModel model, MigrationCommandListBuilder builder)
>    at Microsoft.EntityFrameworkCore.Migrations.MigrationsSqlGenerator.Generate(IReadOnlyList`1 operations, IModel model)
>    at Microsoft.EntityFrameworkCore.Migrations.JetMigrationsSqlGenerator.Generate(IReadOnlyList`1 operations, IModel model)
>    at Microsoft.EntityFrameworkCore.Migrations.Internal.Migrator.GenerateUpSql(Migration migration)
>    at Microsoft.EntityFrameworkCore.Migrations.Internal.Migrator.<>c__DisplayClass15_2.<GetMigrationCommandLists>b__2()
>    at Microsoft.EntityFrameworkCore.Migrations.Internal.Migrator.<MigrateAsync>d__14.MoveNext()
>    at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
>    at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
>    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
>    at System.Runtime.CompilerServices.TaskAwaiter.GetResult()

This was with:
**Microsoft.EntityFrameworkCore 3.1.15
 EntityFrameworkCore.Jet 3.1.0**

**Comment by lauxjpn** [comment]
@lcrumbling Thanks for reporting this issue!

The fix will be part of the upcoming `3.1.1` service release. Until then, just use the lastest [daily build](https://github.com/bubibubi/EntityFrameworkCore.Jet#daily-builds).

**Comment by lcrumbling** [comment]
> The fix will be part of the upcoming `3.1.1` service release. Until then, just use the lastest [daily build](https://github.com/bubibubi/EntityFrameworkCore.Jet#daily-builds).

That's awesome! Thanks for looking at this (and resolving it!) so quickly :)

- Lynn

**Comment by lauxjpn** [comment]
> Does something need to happen in order to expose the daily builds to Nuget?

Ah, looks like I forgot to bump-up the version.
The last one from yesterday (or just the newest one in general) is correct in this case, which should be:

`3.1.0-rtm.1.ci.20210524053456+sha.db2c056`

But I will bump-up the version now as well, so a CI prerelease with the correct version should be available in about 20 minutes.
